ls1intum/Artemis
# Feature/user statistics periods

### Checklist

### Motivation and Context


New features for the user statistics page

### Description

- decoupled the graph from the statistics component to prevent duplicated code with multiple graphs
- removed deprecated code as this was highly confusing in the last PR
- added the arrow functionality + 2 new graphs (active users, released exercises).
- Active users describe the amount of distinct users (Not including test users), which created a submission. Therefore, after uploading the first submission, the second submission in the same timeslot should not make a difference

### Steps for Testing


1. Log in to Artemis
2. Navigate to User statistics ( ->Server Administration -> User statistics)
3. create new Exercises with ReleaseDates and submit solutions
4. Look in the charts that the correct bar increased by one
(You might choose releaseDates near 00:00 which are prone to errors)
5. Try out the arrows and check that the values are correct as well as the description below the graph

### Screenshots



The initial start-view display the last week:
![Screenshot 2020-12-16 at 09 15 58](https://user-images.githubusercontent.com/44401560/102322926-1aa02a00-3f80-11eb-9d45-8d5252eaa732.png)

It is possible to switch timeslots of graphs independently from each other:
![Screenshot 2020-12-16 at 09 17 40](https://user-images.githubusercontent.com/44401560/102323058-4a4f3200-3f80-11eb-9609-f632b7246dd2.png)

There are 4 tabs displaying 4 different timespans, for example Year Tab:
![Screenshot 2020-12-16 at 09 19 18](https://user-images.githubusercontent.com/44401560/102323137-62bf4c80-3f80-11eb-8768-a6ef451e3fe1.png)
